compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -1152,7 +1152,11 @@ public BLangNode transform(ObjectTypeDescriptorNode objTypeDescNode) {
             return objectTypeNode;
         }
 
-        return deSugarTypeAsUserDefType(objectTypeNode);
+        BLangType objectTypeBLangNode = objectTypeNode;
```

Comment:
We need to upcast `BLangObjectTypeNode ` to `BLangType`.
Otherwise https://github.com/ballerina-platform/ballerina-lang/blob/893595abf0bc8dd4f0933b591f0c6563baa6c209/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java#L2076 will not evaluated as true.

I will check it again

Reply:
Updated b57f9c2e9be09e26768f6348dffc5ba2c1984432